mmv1/products/networkservices/GrpcRoute.yaml

```diff
@@ -0,0 +1,249 @@
+                properties:
+                  - !ruby/object:Api::Type::String
+                    name: serviceName
+                    description: |
+                      The URL of a BackendService to route traffic to.
+                  - !ruby/object:Api::Type::Integer
+                    name: weight
+                    description: |
+                      Specifies the proportion of requests forwarded to the backend referenced by the serviceName field.
+            - !ruby/object:Api::Type::NestedObject
+              name: faultInjectionPolicy
+              description: |
+                The specification for fault injection introduced into traffic to test the resiliency of clients to backend service failure.
+              properties:
+                - !ruby/object:Api::Type::NestedObject
+                  name: delay
+                  description: |
+                    Specification of how client requests are delayed as part of fault injection before being sent to a destination.
+                  properties:
+                    - !ruby/object:Api::Type::String
+                      name: fixedDelay
+                      description: |
+                       Specify a fixed delay before forwarding the request.
+                    - !ruby/object:Api::Type::Integer
+                      name: percentage
+                      description: |
+                         The percentage of traffic on which delay will be injected.
+                - !ruby/object:Api::Type::NestedObject
+                  name: abort
+                  description: |
+                     Specification of how client requests are aborted as part of fault injection before being sent to a destination.
+                  properties:
+                    - !ruby/object:Api::Type::Integer
+                      name: httpStatus
+                      description: |
+                         The HTTP status code used to abort the request.
+                    - !ruby/object:Api::Type::Integer
+                      name: percentage
+                      description: |
+                         The percentage of traffic which will be aborted.
+            - !ruby/object:Api::Type::String
+              name: timeout
+              description: |
+                Specifies the timeout for selected route.
+            - !ruby/object:Api::Type::NestedObject
+              name: retryPolicy
+              description: |
+                Specifies the retry policy associated with this route.
+              properties:
+                - !ruby/object:Api::Type::Array
+                  name: retryConditions
+                  description: |
+                    Specifies one or more conditions when this retry policy applies.
+                  item_type: !ruby/object:Api::Type::Enum
+                      name: retryCondition
+                      description: Condition when this retry policy applies.
+                      values:
```

Comment:
Do we expect more values for this to be added in the future? A string value with a list or link to a description of the possible values may be better if we expect new values to be introduced.

Reply:
I dont think there are plans to add new values. However, we should be able to update in Enum in the future with new values as well?

Reply:
I see your point, if this is a string then there would be no need to update the TF module.
However, these values are pretty static and the proto hasn't changed in more than 2 years, so I think we should be good.